File: sekre-backend/internal/domain/entity/division.go
package entity

import (
	"time"

	"github.com/google/uuid"
	"github.com/username/sekre-backend/internal/domain/types"
)

// Division represents organizational division - pure business entity
type Division struct {
	ID             uuid.UUID `json:"id"`
	OrganizationID uuid.UUID `json:"organization_id"`
	Name           string    `json:"name"`
	Description    *string   `json:"description,omitempty"`
	CreatedAt      time.Time `json:"created_at"`
	UpdatedAt      time.Time `json:"updated_at"`
}

// DivisionMember represents member in division
type DivisionMember struct {
	DivisionID   uuid.UUID          `json:"division_id"`
	UserID       uuid.UUID          `json:"user_id"`
	DivisionRole types.DivisionRole `json:"division_role"`
	JoinedAt     time.Time          `json:"joined_at"`
}

// DivisionWithMembers combines division + members
type DivisionWithMembers struct {
	Division Division       `json:"division"`
	Members  []UserWithRole `json:"members"`
}

// UserWithRole combines user + division role
type UserWithRole struct {
	User         User               `json:"user"`
	DivisionRole types.DivisionRole `json:"division_role"`
	JoinedAt     time.Time          `json:"joined_at"`
}

// Invitation represents member invitation
type Invitation struct {
	ID             uuid.UUID              `json:"id"`
	OrganizationID uuid.UUID              `json:"organization_id"`
	Email          string                 `json:"email"`
	Token          string                 `json:"token"`
	Status         types.InvitationStatus `json:"status"`
	InvitedBy      uuid.UUID              `json:"invited_by"`
	ExpiresAt      time.Time              `json:"expires_at"`
	CreatedAt      time.Time              `json:"created_at"`
}

File: sekre-mobile/composeApp/src/commonMain/kotlin/org/sekre_mobile/com/domain/entity/Division.kt
package org.sekre_mobile.com.domain.entity

/** Division Domain Entity Pure business object - no framework dependencies */
data class Division(
    val id: String,
    val organizationId: String,
    val name: String,
    val createdAt: Long,
    val updatedAt: Long
) {
    /** Check if division can be deleted */
    fun canBeDeleted(): Boolean = true

    /** Check if division can be edited */
    fun canBeEdited(): Boolean = true
}

enum class DivisionRole {
    HEAD,
    STAFF
}

data class DivisionMember(
    val divisionId: String,
    val userId: String,
    val divisionRole: DivisionRole
)
